pkg/daemon/plugin.go

```diff
@@ -29,61 +29,91 @@ var (
 	K8sPlugin         = k8splugin.NewK8sPlugin
 )
 
-func enablePlugins(platform utils.PlatformType, useSystemdService bool, ns *sriovnetworkv1.SriovNetworkNodeState, hostManager host.HostManagerInterface, storeManager utils.StoreManagerInterface) (map[string]plugin.VendorPlugin, error) {
-	log.Log.Info("enableVendorPlugins(): enabling plugins")
-	enabledPlugins := map[string]plugin.VendorPlugin{}
+func loadPlugins(
+	platform utils.PlatformType,
+	useSystemdService bool,
+	ns *sriovnetworkv1.SriovNetworkNodeState,
+	hostManager host.HostManagerInterface,
+	storeManager utils.StoreManagerInterface,
+	disabledPlugins []string) (map[string]plugin.VendorPlugin, error) {
+	log.Log.Info("loadPlugins(): loading plugins")
+	loadedPlugins := map[string]plugin.VendorPlugin{}
 
 	if platform == utils.VirtualOpenStack {
 		virtualPlugin, err := VirtualPlugin(false)
 		if err != nil {
-			log.Log.Error(err, "enableVendorPlugins(): failed to load the virtual plugin")
+			log.Log.Error(err, "loadPlugins(): failed to load the virtual plugin")
 			return nil, err
 		}
-		enabledPlugins[virtualPlugin.Name()] = virtualPlugin
+		pluginName := virtualPlugin.Name()
+		if !isPluginDisabled(pluginName, disabledPlugins) {
+			loadedPlugins[pluginName] = virtualPlugin
+		}
 	} else {
-		enabledVendorPlugins, err := registerVendorPlugins(ns)
+		loadedVendorPlugins, err := loadVendorPlugins(ns, disabledPlugins)
```

Comment:
I think we should only allow to disable vendor plugins (the ones we loaded with the `loadVendorPlugins`). The others are essential for the daemon to work. Virtual for openstack and generic for baremetal. It doesn't make sense to allow users to disable them, unless you see a valid use case.

Reply:
> It doesn't make sense to allow users to disable them, unless you see a valid use case.

ATM this is blocked at API level (user can only disable a single plugin atm), from code perspective, it seems easier to make reason about. 

i.e daemon calls "loadPlugins" and it has an arg with a list of which plugins are disabled.

possible use-case for disabling generic plugin - all sriov configurations are preconfigured and we just want the operator to handle "the rest"
i.e device plugin and its configuration,  cni deployment, net-attach-def creation

that said i dont have such a use-case atm. my main argument is less cognitive complexity.

Reply:
fair enough